lib/fluent/plugin/out_pgjson.rb

```diff
@@ -94,10 +94,16 @@ def write(chunk)
         errmsg = "%s while copy data: %s" % [ err.class.name, err.message ]
         @conn.put_copy_end( errmsg )
         @conn.get_result
+        @conn.close()
```

Comment:
why the line 87 of `@conn.exec` not include in the `begin`? It may also raise exception but will not protected by the following `rescue`
> Sends SQL query request specified by sql to PostgreSQL. On success, it returns a PG::Result instance with all result rows and columns. On failure, it raises a PG::Error.

from https://deveiate.org/code/pg/PG/Connection.html

Reply:
And if the connection lost, could `@conn.get_result` raise another un-caught exception?

Reply:
@mydmdm I agree that @conn.exec might throw an exception and that the current code does not seem to take that into account

With regards to the pull request though I see that as a separate issue, because even before the pull request, that error would have been in there.  And without the pull request its still broken - only more or less so.   The scope of the pullrequest was limited to reinitialize the connection to postgres in case it timed out.

Reply:
Are we expecting a merge anytime soon? That would help me a lot. :)